src/js/bookmarksPaginator.js

```diff
@@ -0,0 +1,295 @@
+/*jslint node: true */
+'use strict';
+let _ = require('lodash');
+let constants = require("./constants");
+let common = require('./common');
+
+/**
+ * Wrapper around ordinary request. Sends request with prefilled access token.
+ * @param url String API endpoint
+ * @param data JSON
+ * @returns {Promise}
+ */
+function request(url, data) {
```

Comment:
this is the only function that sends requests, so we only need one message to background script...